control/controlhttp/client_js.go

```diff
@@ -16,21 +16,27 @@ import (
 	"tailscale.com/types/key"
 )
 
-// Variant of Dial that tunnels the request over WebScokets, since we cannot do
+// Variant of Dial that tunnels the request over WebSockets, since we cannot do
 // bi-directional communication over an HTTP connection when in JS.
 func Dial(ctx context.Context, addr string, machineKey key.MachinePrivate, controlKey key.MachinePublic, protocolVersion uint16, dialer dnscache.DialContextFunc) (*controlbase.Conn, error) {
 	init, cont, err := controlbase.ClientDeferred(machineKey, controlKey, protocolVersion)
 	if err != nil {
 		return nil, err
 	}
 
-	host, addr, err := net.SplitHostPort(addr)
+	host, port, err := net.SplitHostPort(addr)
 	if err != nil {
 		return nil, err
 	}
+	wsScheme := "wss"
+	wsHost := host
```

Comment:
and this is just to make 80 into 443, right?

related: https://github.com/tailscale/tailscale/pull/4323

cc @maisem

Reply:
nevermind, this is client_js.go

Reply:
I think client_js.go could be folded into client.go (as a third fallback to HTTP/HTTPS) if we wanted to do that bug and/or tailscale/corp#5542.